pkg/controller/importconfig/import_worker.go

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package importconfig
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/stolostron/managedcluster-import-controller/pkg/constants"
+	"github.com/stolostron/managedcluster-import-controller/pkg/features"
+	"github.com/stolostron/managedcluster-import-controller/pkg/helpers"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	operatorv1 "open-cluster-management.io/api/operator/v1"
+)
+
+type importWorker interface {
+	generateImportSecret(ctx context.Context, managedCluster *clusterv1.ManagedCluster) (*corev1.Secret, error)
+}
+
+type workerFactory struct {
+	clientHolder *helpers.ClientHolder
+}
+
+func (f *workerFactory) newWorker(mode string) (importWorker, error) {
```

Comment:
cool! !
nit: could we separate this in different files and org them in a dir, like
```
importconfig
  |--works
         |-factory.go
         |-defualt_work.go
         |-detach_work.go
```

Reply:
Have put the workers in multiple files, but did move them to another folder since the global `log` variable is hard to process.